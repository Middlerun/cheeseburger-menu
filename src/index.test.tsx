import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'

import CheeseburgerMenu from './index'

const emptyFunc = (): null => null

describe('<CheeseburgerMenu/>', () => {
  it('matches snapshot when closed', () => {
    const closedMenu = ReactShallowRenderer.createRenderer().render(<CheeseburgerMenu isOpen={false} closeCallback={emptyFunc}>content</CheeseburgerMenu>)
    expect(closedMenu).toMatchSnapshot()
  })

  it('matches snapshot when open', () => {
    const openMenu = ReactShallowRenderer.createRenderer().render(<CheeseburgerMenu isOpen={true} closeCallback={emptyFunc}>content</CheeseburgerMenu>)
    expect(openMenu).toMatchSnapshot()
  })
})
