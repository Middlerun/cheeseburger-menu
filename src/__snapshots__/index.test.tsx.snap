// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheeseburgerMenu/> matches snapshot when closed 1`] = `
<div
  className="cheeseburger-menu"
>
  <div
    className="cheeseburger-menu-overlay"
    onClick={[Function]}
    style={
      {
        "background": "rgba(0, 0, 0, 0.3)",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "transform": "translate3d(100%, 0px, 0px)",
        "transition": "opacity 0.3s, transform 0s 0.3s",
        "zIndex": 1001,
      }
    }
  />
  <ReactSwipe
    allowMouseEvents={false}
    innerRef={[Function]}
    onSwipeDown={[Function]}
    onSwipeEnd={[Function]}
    onSwipeLeft={[Function]}
    onSwipeMove={[Function]}
    onSwipeRight={[Function]}
    onSwipeStart={[Function]}
    onSwipeUp={[Function]}
    tagName="div"
    tolerance={0}
  >
    <div
      className="cheeseburger-menu-outer"
      style={
        {
          "backgroundColor": "white",
          "bottom": 0,
          "left": 0,
          "maxWidth": "80%",
          "position": "fixed",
          "right": "inherit",
          "top": 0,
          "transform": "translate3d(-100%, 0px, 0px)",
          "transformOrigin": "left",
          "transition": "transform 0.3s",
          "width": 300,
          "zIndex": 1002,
        }
      }
    >
      <div
        className="cheeseburger-menu-inner"
        style={
          {
            "height": "100%",
            "overflowY": "auto",
            "paddingBottom": 0,
          }
        }
      >
        content
      </div>
      <div
        className="cheeseburger-menu-shadow"
        style={
          {
            "bottom": 0,
            "boxShadow": "0 0 15px 0 rgba(0, 0, 0, .2)",
            "height": "100%",
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transition": "opacity 0.3s",
            "width": "100%",
            "zIndex": -1,
          }
        }
      />
    </div>
  </ReactSwipe>
</div>
`;

exports[`<CheeseburgerMenu/> matches snapshot when open 1`] = `
<div
  className="cheeseburger-menu open"
>
  <div
    className="cheeseburger-menu-overlay"
    onClick={[Function]}
    style={
      {
        "background": "rgba(0, 0, 0, 0.3)",
        "bottom": 0,
        "left": 0,
        "opacity": 1,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "transform": "none",
        "transition": "opacity 0.3s",
        "zIndex": 1001,
      }
    }
  />
  <ReactSwipe
    allowMouseEvents={false}
    innerRef={[Function]}
    onSwipeDown={[Function]}
    onSwipeEnd={[Function]}
    onSwipeLeft={[Function]}
    onSwipeMove={[Function]}
    onSwipeRight={[Function]}
    onSwipeStart={[Function]}
    onSwipeUp={[Function]}
    tagName="div"
    tolerance={0}
  >
    <div
      className="cheeseburger-menu-outer"
      style={
        {
          "backgroundColor": "white",
          "bottom": 0,
          "left": 0,
          "maxWidth": "80%",
          "position": "fixed",
          "right": "inherit",
          "top": 0,
          "transform": "translate3d(0px, 0px, 0px)",
          "transformOrigin": "left",
          "transition": "transform 0.3s",
          "width": 300,
          "zIndex": 1002,
        }
      }
    >
      <div
        className="cheeseburger-menu-inner"
        style={
          {
            "height": "100%",
            "overflowY": "auto",
            "paddingBottom": 0,
          }
        }
      >
        content
      </div>
      <div
        className="cheeseburger-menu-shadow"
        style={
          {
            "bottom": 0,
            "boxShadow": "0 0 15px 0 rgba(0, 0, 0, .2)",
            "height": "100%",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transition": "opacity 0.3s",
            "width": "100%",
            "zIndex": -1,
          }
        }
      />
    </div>
  </ReactSwipe>
</div>
`;
